src/hacking/building-and-running-on-wsl.md

```diff
@@ -0,0 +1,43 @@
+# Building and Running on WSL
+
+To run non-headless servo shell on WSL, you will most likely need to be on Windows 10 Build 19044+ or Windows 11 as access to WSL v2 is needed for GUI.
+
+## Building
+
+Servo can be built on WSL as if you are building on any other Linux distribution.
+
+1. Setup WSL v2.  See [Microsoft's guideliens for setting up GUI apps in WSL](https://learn.microsoft.com/en-us/windows/wsl/tutorials/gui-apps).
+2. Setup the environment depending on the WSL distribution that you are using (e.g. Ubuntu, OpenSuse, etc)
+3. Build
+
+## Running
+
+WSL v2 has the corresponding adaptors to display wayland and x11 applications, though it may not always work OOB with servo.
```

Comment:
```suggestion
WSL v2 has the corresponding adaptors to display Wayland and X11 applications, though it may not always work out of the box with Servo.
```